common/lib/common-definitions/src/logger.ts

```diff
@@ -120,3 +120,23 @@ export interface ITelemetryLogger extends ITelemetryBaseLogger {
      */
     shipAssert(condition: boolean, event?: ITelemetryErrorEvent): void;
 }
+
+// ////////////////////// fluidError.ts in common-definitions /////////////////////////////
+/**
+ * Can be decl merged to add stuff like packagename, odspErrorResponse, etc
+ * The logger impl can choose to log anything in here as appropriate, but by default it's not
+ */
+export interface ISensitiveDebugData {
+    innerError?: any;
+}
+
+export interface IFluidError {
+    errorType: string;
+    message: string;
+    getFluidTelemetryProps: () => ITelemetryProperties;  //* use property getters?
+    getSensitiveDebugData: () => ISensitiveDebugData & { stack: string };
+    addDetails: (props: ITelemetryProperties, debugData: ISensitiveDebugData) => void;
```

Comment:
i'm not a fan of addDetails here. i perfer to keep the interface and simple as possible

Reply:
There are a few places I've noticed in our code where we're doing `(error as any).foo = whatever`. I'd like to give a better way to do that.